server-side/controllers/user.controller.js

```diff
@@ -30,60 +37,189 @@ export const getUserById = async (req, res,next) => {
+//       });
+//     }
+//   } catch (error) {
+//     console.error(messages.error.G_LOGIN_FAILED, error);
+//     res.status(500).json({ error: messages.error.G_LOGIN_FAILED });
+//   }
+// };
+export const getUserByGoogleAccount = async (req, res) => {
+  const clientId = process.env.CLIENT_ID;
+  const secretCode = process.env.SECRET_CODE;
+
+  // קבלת הטוקן מהכותרת של הבקשה
+  const token = req.headers['authorization']?.split(' ')[1]; // מצפים לקבל את הטוקן בכותרת Authorization: Bearer <token>
+
+  if (!token) {
+    return res.status(400).json({ error: 'Token is required' });
+  }
 
-export const deleteUser = async (req, res,next) => {
-  const id = req.params.id;
-  if(!mongoose.Types.ObjectId.isValid(id))
-    return next({message:'id is not valid'})
   try {
-    const user = await Users.findByIdAndDelete(id);
-    if (!user) {
-      return next({message:'user not found ',status:404})
+    // Verify the token with Google
+    console.log("=========================================================================================================================================================")
+    
+    const ticket = await client.verifyIdToken({
+        idToken: token,
+      audience: clientId
+    });
+    console.log("sfdghj")
+    
+    // Get user information from token payload
+    const payload = ticket.getPayload();
+    const { email, name, picture } = payload;
+    
+    // Check if the user exists in your database
+    const user = await getByEmail1(email);
+    
+    if (user) {
+      return res.status(200).send(user);
+    } else {
+      // Create a JWT token to send back to the client
+      const newToken = jwt.sign({ email, name, picture }, secretCode, { expiresIn: '1h' });
       
+      // Respond with user data and JWT token
+      res.json({
+        token: newToken,
+        userData: { email, name, picture }
+      });
     }
-    res.send('User deleted successfully!');
-  } catch (err) {
-    console.error(err);
-    next({message:err.message,status:500})
+  } catch (error) {
+    console.error('Error verifying Google token:', error);
+    res.status(500).json({ error: 'Failed to verify Google token' });
   }
 };
 
-export const updatedUser = async (req, res,next) => {
-  const id = req.params.id;
-  if(!mongoose.Types.ObjectId.isValid(id))
-    return next({message:'id is not valid'})
+  export const getByEmail = async (req, res) =>
+    {
+      try{
+        const email= req.params;
+        const user = await User.findOne({ email: email});
+       if(user)
+        res.status(200).send(user)
+      }
+      catch(error){
+      console.error(error);
+      res.status(404).send(messages.error.EMAIL_NOT_FOUND);
+     }
+    }
+  export const getByEmail1 = async (email) => {
+              const user = await User.findOne({ email });
+              return user; 
+      };
+  export const getCode = async (req, res)=>{
+    const {email}=req.params.email;
+    const password= req.params.password;
```

Comment:
please use Destructuring Assignment  in each place in your code that it's enabled. for example here - for the password